### PR TITLE
Update docs on Compose environment files

### PR DESCRIPTION
**Pull request by aiordache**
Reorganize sections for variable substitution and document `.env` file path change for `1.28`.

### CODE REVIEW

File: compose/env-file.md [hunk 1]
```diff
@@ -5,8 +5,10 @@ title: Declare default environment variables in file
 ---
 
 Compose supports declaring default environment variables in an environment file
-named `.env` placed in the project directory. The project directory is specified by order of precedence:
-- `--project-dir` flag
+named `.env` placed in the project directory. Versions of Docker Compose prior to `1.28`, load the `.env` from the current working directory where the command is executed, or, the project directory if this is explicitly set with the `--project-directory` argument. This inconsistency has been addressed since `+v1.28` by limiting the `.env` file path to the project directory.
+
```
Comment (ndeloof) [inline compose/env-file.md:9]: you could add here a note that users relying on this inconsistent behaviour have a simple workaround by using --env-file explicitly

File: compose/env-file.md [hunk 2]
```diff
@@ -5,8 +5,10 @@ title: Declare default environment variables in file
...
+The project directory is specified by order of precedence:
+- `--project-directory` flag
 - Folder of the first `--file` flag
 - Current directory
```
Comment (ndeloof) [inline compose/env-file.md:13]: I'm surprised docker-compose doesn't have support for some COMPOSE_PROJECT_DIR env variable to be even more confusing as it use to be :P

**Review by ndeloof — DISMISSED** [review]

File: compose/environment-variables.md [hunk 3]
```diff
@@ -21,10 +21,66 @@ web:
...
+Compose file, or used to configure Compose, in an [environment file](env-file.md)
+named `.env`. The `.env` file path is as follows:
+  - Since `+v1.28`, `.env.` file is placed at the base of the project directory 
+  - For previous versions, it is placed in the current working directory where the
+  Docker Compose command is executed unless a --project-directory is defined which
+  overrides the path for the `.env` file. This inconsistency has been addressed
+  in `+v1.28` by limiting the filepath to the project directory.
+
+
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:41]: nit:
```suggestion
```shell
```

File: compose/environment-variables.md [hunk 4]
```diff
@@ -21,10 +21,66 @@ web:
...
+  web:
+    image: "webapp:${TAG}"
+```
+
+When you run `docker-compose up`, the `web` service defined above uses the
+image `webapp:v1.5`. You can verify this with the
+[config command](reference/config.md), which prints your resolved application
+config to the terminal:
+
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:57]:
```suggestion
```shell
```

File: compose/environment-variables.md [hunk 5]
```diff
@@ -21,10 +21,66 @@ web:
...
+  web:
+    image: 'webapp:v1.5'
+```
+
+Values in the shell take precedence over those specified in the `.env` file.
+
+If you set `TAG` to a different value in your shell, the substitution in `image`
+uses that instead:
+
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:71]:
```suggestion
```shell
```

File: compose/environment-variables.md [hunk 6]
```diff
@@ -33,6 +89,48 @@ done using the `--env-file` option:
...
+$ docker-compose config 
+version: '3'
+services:
+  web:
+    image: 'webapp:v1.5'
+```
+Passing the `--env-file ` argument overrides the default file path:
+
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:120]:
```suggestion
```shell
```

File: compose/environment-variables.md [hunk 7]
```diff
@@ -33,6 +89,48 @@ done using the `--env-file` option:
...
+
+$ cat docker-compose.yml
+version: '3'
+services:
+  web:
+    image: "webapp:${TAG}"
+```
+
+The `.env` file is loaded by default:
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:111]:
```suggestion
```shell
```

File: compose/environment-variables.md [hunk 8]
```diff
@@ -33,6 +89,48 @@ done using the `--env-file` option:
 ```shell
 docker-compose --env-file ./config/.env.dev up 
 ```
+This file path is relative to the current working directory where the Docker Compose
+command is executed.
+
+```bash
```
Comment (ulyssessouza) [inline compose/environment-variables.md:95]:
```suggestion
```shell
```

**Review by ulyssessouza — DISMISSED** [review]
LGTM with @ndeloof 's suggestion

**Review by ndeloof — APPROVED** [review]

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**